### PR TITLE
[SM-582] Adapt documentation for policies

### PR DESCRIPTION
**Pull request by NachoRaik**

### CODE REVIEW

File: modules/ROOT/pages/client-id-based-policies.adoc [hunk 1]
```diff
@@ -157,6 +158,40 @@ When an HTTP request is performed against a protected resource for which the Cli
 
 For SOAP applications, the response is a SOAP error.
 
+== Service Mesh Basic Version
+
+For Service Mesh, there is a variety of this policy, named Basic Client ID Enforcement Policy. Although the functionality is similar to the non-basic kind, there are differences in the policy configuration:
```
Comment (svacas) [inline modules/ROOT/pages/client-id-based-policies.adoc]: to be more consistent with the rest of the policies I'd go with
```
The Basic ... Policy works in the same way as the regular policy, with the following limitations...
```

File: modules/ROOT/pages/client-id-based-policies.adoc [hunk 2]
```diff
@@ -157,6 +158,40 @@ When an HTTP request is performed against a protected resource for which the Cli
...
+
+|Client ID Header
+| The header name to be used to extract the client ID from API requests.
+| Yes.
+
+|Client Secret Header
+| The header name to be used to extract the client Secret from API requests
+| No.
+
+|Method & Resource conditions: Apply configurations to all API methods & resources
```
Comment (svacas) [inline modules/ROOT/pages/client-id-based-policies.adoc]: resource level config shouldn't be here as is part of the regular policy

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc [hunk 3]
```diff
@@ -185,6 +185,88 @@ The following table provides instructions for how to convert your existing key b
 . The public key will be in the file public.pem
 |===
 
+== Service Mesh Basic Version
+
+For Service Mesh, there is a variety of this policy, named Basic JWT Validation Policy. Although the functionality is similar to the non-basic kind, there are differences in the policy configuration:
```
Comment (svacas) [inline modules/ROOT/pages/policy-mule4-jwt-validation.adoc]: Use same pattern as with the rest of the policies
```
The Basic ... Policy works in the same way as the regular policy, with the following limitations...
```

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc [hunk 4]
```diff
@@ -185,6 +185,88 @@ The following table provides instructions for how to convert your existing key b
...
+If you set it to `jwt` Header, the JWT will be extracted from that header. +
+If you set it to HTTP Bearer Authentication Header, the JWT will be expected as Bearer. +
+If you set this field to *Custom Header*, the name of the header where the JWT is expected must be provided.
+|
+
+| Token Header 
+| If you set the JWT Origin to *Custom Header*, type the name of the header containing the JWT here.
+| `myJwtHeader`
+
+| JWT Signing Method
```
Comment (svacas) [inline modules/ROOT/pages/policy-mule4-jwt-validation.adoc]: List only the fields that are configured in a different way or removed

File: modules/ROOT/pages/policy-openid-connect.adoc [hunk 5]
```diff
@@ -94,6 +94,13 @@ The fields returned by the OAuth provider are processed by the policy, and:
 
 TIP: You can use the extra information to further tune your business logic.
 
+=== Open ID Policy for Service Mesh
+
+The Basic Open ID Connect OAuth 2.0 Token Enforcement Policy works in the same way as the regular policy, with the following differences:
+
+* Does not offer the option to expose headers
+* Does not include the option to enable third-party authentication server TLS validation, as the validation would occur from the edge of the cluster, not from the Mule Runtime pod
```
Comment (svacas) [inline modules/ROOT/pages/policy-openid-connect.adoc]:
```suggestion
* Does not include the option to enable third-party authentication server TLS validation using self-signed certificates
```

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc [hunk 6]
```diff
@@ -208,63 +211,10 @@ If you set this field to *Custom Header*, the name of the header where the JWT i
...
-
-| Validate Expiration Claim
-| Indicates that the policy should check for the validity of the expiration claim. You can set this claim as "Mandatory" by selecting *Expiration Claim Mandatory*.
-|
-
-| Validate Not Before Claim
-| Indicates that the policy should check for the validity of the Not Before claim. You can set this claim as "Mandatory" by selecting *Not Before Claim Mandatory*
-|
-
-| Validate Custom Claim
```
Comment (svacas) [inline modules/ROOT/pages/policy-mule4-jwt-validation.adoc]: custom claim does not support DW, so it should be included

**Review by svacas — APPROVED** [review]

**Review by tbruno — APPROVED** [review]
Good Job @NachoRaik . Thanks for taking care of this!

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc [hunk 7]
```diff
@@ -71,9 +71,13 @@ To access the claim values you can use the following Dataweave 2.0 expression by
 
 == Configuring Policy Parameters
 
-When you apply the policy to your API from the UI, the following parameters are displayed:
+When you apply the policy to your API from the UI, a list of parameters are displayed based on whether your environment includes Mule or non-Mule applications.
 
-[%header%autowidth.spread,cols="a,a,a"]
+=== Confiuring the Parameters for Mule Applications
```
Comment (NachoRaik) [inline modules/ROOT/pages/policy-mule4-jwt-validation.adoc]:
```suggestion
=== Configuring the Parameters for Mule Applications
```

File: modules/ROOT/pages/client-id-based-policies.adoc [hunk 8]
```diff
@@ -116,21 +116,26 @@ curl "http://localhost/myResource" -u 1234:abcd
...
+| Specifies from where in the request to extract the values:
 
-Custom Expression: Accepts an expression each for client ID and client secret, indicating where to extract the credentials from the request. For Mule 3, use MEL Expressions and for Mule 4 you must provide a DataWeave 2.0 expression. Use this option to send the credentials in a custom header or somewhere else in the payload. You can also configure the client secret requirement as optional by leaving the field blank.
+* HTTP Basic Authentication Header: Requires credentials as part of the authorization header. The application consuming the API must use the basic authentication scheme to send the credentials in the requests.
+* Custom Expression: Accepts an expression each for client ID and client secret, indicating where to extract the credentials from the request. For Mule 3, use MEL Expressions and for Mule 4, use a DataWeave 2.0 expression. Use this option to send the credentials in the payload in a custom header or any other format. You can leave the field blank to configure the client secret requirement as optional.
```
Comment (NachoRaik) [inline modules/ROOT/pages/client-id-based-policies.adoc]:
```suggestion
* Custom Expression: Accepts an expression each for client ID and client secret, indicating where to extract the credentials from the request. For Mule 3, use MEL Expressions and for Mule 4, use a DataWeave 2.0 expression. Use this option to send the credentials in the payload, in a custom header or any other format. You can leave the field blank to configure the client secret requirement as optional.
```

File: modules/ROOT/pages/rate-limiting-and-throttling-sla-based-policies.adoc [hunk 9]
```diff
@@ -76,34 +76,34 @@ Within the next 19100 milliseconds, only 14 more requests are allowed by the SLA
 
 The guidelines for rate limiting in a cluster applies to rate limiting for SLA-based policies.
 
-=== Rate Limiting SLA for Service Mesh
+=== Rate Limiting SLA for Anypoint Service Mesh
 
-The Basic Rate Limiting SLA Policy works in the same way as the regular policy, with the following differences:
+The Rate Limiting SLA policy works in the same way for Anypoint Service Mesh (non-Mule applications), excluding the following differnces:
```
Comment (NachoRaik) [inline modules/ROOT/pages/rate-limiting-and-throttling-sla-based-policies.adoc]:
```suggestion
The Rate Limiting SLA policy works in the same way for Anypoint Service Mesh (non-Mule applications), excluding the following differences:
```

File: modules/ROOT/pages/client-id-based-policies.adoc (hunk 8 above)
Comment (priyaayyar) [reply modules/ROOT/pages/client-id-based-policies.adoc]: Done! Thanks!

File: modules/ROOT/pages/policy-mule4-jwt-validation.adoc (hunk 7 above)
Comment (priyaayyar) [reply modules/ROOT/pages/policy-mule4-jwt-validation.adoc]: Done! Thanks!

File: modules/ROOT/pages/rate-limiting-and-throttling-sla-based-policies.adoc (hunk 9 above)
Comment (priyaayyar) [reply modules/ROOT/pages/rate-limiting-and-throttling-sla-based-policies.adoc]: Done! Thanks!

**Review by priyaayyar — APPROVED** [review]

**Pull request merged**